lib/vrl/stdlib/src/hmac.rs

```diff
@@ -0,0 +1,226 @@
+use ::value::Value;
+use hmac::{Hmac as HmacHasher, Mac};
+use sha1::Sha1;
+use sha_2::{Sha224, Sha256, Sha384, Sha512};
+use vrl::prelude::*;
+
+macro_rules! hmac {
+    ($algorithm:ty, $key:expr, $val:expr) => {{
+        let mut mac = <HmacHasher<$algorithm>>::new_from_slice($key.as_ref()).expect("key is bytes");
+        mac.update($val.as_ref());
+        let result = mac.finalize();
+        let code_bytes = result.into_bytes();
+        code_bytes.to_vec()
+    }};
+}
+
+fn hmac(value: Value, key: Value, algorithm: &Bytes, encoding: &Bytes) -> Resolved {
+    let value = value.try_bytes()?;
+    let key = key.try_bytes()?;
+
+    let code_bytes = match algorithm.as_ref() {
+        b"SHA1" => hmac!(Sha1, key, value),
+        b"SHA-224" => hmac!(Sha224, key, value),
+        b"SHA-256" => hmac!(Sha256, key, value),
+        b"SHA-384" => hmac!(Sha384, key, value),
+        b"SHA-512" => hmac!(Sha512, key, value),
+        _ => unreachable!("enum invariant")
+    };
+
+    let hash = match encoding.as_ref() {
+        b"hex" => hex::encode(code_bytes),
+        b"base64" => base64::encode(code_bytes),
+        _ => unreachable!("enum invariant")
+    };
+    Ok(hash.into())
+}
+
+fn encode_formats() -> Vec<Value> {
+    vec![
+        value!("hex"),
+        value!("base64")
+    ]
+}
+
+fn algorithms() -> Vec<Value> {
+    vec![
+        value!("SHA1"),
+        value!("SHA-224"),
+        value!("SHA-256"),
+        value!("SHA-384"),
+        value!("SHA-512"),
+    ]
+}
+
+#[derive(Clone, Copy, Debug)]
+pub struct Hmac;
+
+impl Function for Hmac {
+    fn identifier(&self) -> &'static str {
+        "hmac"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[
+            Parameter {
+                keyword: "value",
+                kind: kind::ANY,
+                required: true,
+            },
+
+            Parameter {
+                keyword: "key",
+                kind: kind::ANY,
+                required: true,
+            },
+
+            Parameter {
+                keyword: "algorithm",
+                kind: kind::ANY,
+                required: false
+            },
+
+            Parameter {
+                keyword: "encoding",
+                kind: kind::ANY,
+                required: false
+            }

```

Comment:
Given that you try to convert these the key/value parameters to a bytes representation on lines 18-19 -- where that function basically will either unwrap the value if it's already a bytes value, or throw an error otherwise -- as well as needing the algorithm/encoding to also be bytes... it seems like you should just declare all the parameters as the bytes kind?